common/chat.ts

```diff
@@ -199,3 +241,32 @@ export interface ModerateChannelUserServerRequest {
    */
   moderationReason?: string
 }
+
+/**
+ * Specific chat information about a user in a particular channel, such as their join date, their
+ * channel role (e.g. are they a moderator), etc.
+ */
+export interface ChatUserProfile {
+  joinDate: number
+  /**
+   * User is considered a channel moderator if they have one of the following permissions:
+   *  - owner
+   *  - editPermissions
+   *  - ban
+   *  - kick
+   */
+  isModerator: boolean
+}
+
+export type ChatUserProfileJson = Jsonify<ChatUserProfile>
+
+/**
+ * The response returned when fetching a profile of a user in a specific chat channel. The profile
+ * itself can be `null` in case the user has left the channel, but their name is still visible in
+ * old messages.
+ */
+export interface GetChatUserProfileResponse {
+  userId: SbUserId
+  channelName: string
+  profile: ChatUserProfileJson | null
```

Comment:
In general never use `null` (unless it is part of some existing API we don't control). Vastly, vastly, vastly prefer `undefined` (and optional fields when relevant)